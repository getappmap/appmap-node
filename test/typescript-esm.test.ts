import { integrationTest, readAppmap, runAppmapNode } from "./helpers";

integrationTest("esm-loader is loaded when required", () => {
  expect(runAppmapNode("node", "--loader", "ts-node/esm", "index.ts").status).toBe(0);
  expect(readAppmap()).toMatchSnapshot();
});
