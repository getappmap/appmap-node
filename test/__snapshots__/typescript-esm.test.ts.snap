// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-loader is loaded when required 1`] = `
{
  "classMap": [
    {
      "children": [
        {
          "children": [
            {
              "location": "./index.ts:1",
              "name": "run",
              "static": true,
              "type": "function",
            },
          ],
          "name": "index",
          "type": "class",
        },
      ],
      "name": "index",
      "type": "package",
    },
  ],
  "events": [
    {
      "defined_class": "index",
      "event": "call",
      "id": 1,
      "lineno": 1,
      "method_id": "run",
      "parameters": [],
      "path": "./index.ts",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 2,
      "parent_id": 1,
      "return_value": {
        "class": "Boolean",
        "value": "true",
      },
      "thread_id": 0,
    },
  ],
  "metadata": {
    "app": "appmap-node-typescript-esm-test",
    "client": {
      "name": "appmap-node",
      "url": "https://github.com/getappmap/appmap-node",
      "version": "test node-appmap version",
    },
    "language": {
      "engine": "Node.js",
      "name": "javascript",
      "version": "test node version",
    },
    "name": "test process recording",
    "recorder": {
      "name": "process",
      "type": "process",
    },
  },
  "version": "1.12",
}
`;
